src/main/java/com/iexec/sms/secret/AbstractSecretService.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024-2024 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.iexec.sms.secret;
+
+import lombok.extern.slf4j.Slf4j;
+import net.jodah.expiringmap.ExpirationPolicy;
+import net.jodah.expiringmap.ExpiringMap;
+
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public abstract class AbstractSecretService<K> {
+
+    private final ExpiringMap<K, Boolean> secretExistenceCache = ExpiringMap.builder()
+            .expiration(1, TimeUnit.MINUTES)
+            .expirationPolicy(ExpirationPolicy.CREATED)
+            .build();
+
+    /**
+     * Caches the existence of the secret.
+     *
+     * @param key The key to use for cache
+     */
+    protected void putSecretExistenceInCache(K key, boolean value) {
+        log.debug("Put secret existence in cache[key:{}]", key);
+        if (null != key) {
+            secretExistenceCache.put(key, value);
+        } else {
+            //no strong coupling with cache, no exception handling
+            log.debug("Key is NULL, unable to use cache");
```

Comment:
Isn't this a warning? IMHO, it should be at least a warning. That is not a use case for the cache; we should never try to cache a `null` value.

Reply:
ok, I'll change the level